metricflow/specs/query_interface.py

```diff
@@ -0,0 +1,49 @@
+from __future__ import annotations
+
+from typing import Sequence
+from abc import ABC
+
+
+class QueryInterfaceMetric(ABC):
+    """Metric in the Jinja interface."""
+
+    def __init__(self, name: str) -> None:  # noqa: D
+        raise NotImplementedError
+
+    def pct_growth(self) -> QueryInterfaceMetric:
+        """The percentage growth."""
+        raise NotImplementedError
+
+
+class QueryInterfaceDimension(ABC):
+    """Dimension in the Jinja interface."""
+
+    def __init__(self, name: str, entity_path: Sequence[str] = ()) -> None:  # noqa: D
+        raise NotImplementedError
+
+    def grain(self, _grain: str) -> QueryInterfaceDimension:
+        """The time granularity."""
+        raise NotImplementedError
+
+    def alias(self, _alias: str) -> QueryInterfaceDimension:
+        """Renaming the column."""
+        raise NotImplementedError
+
+
+class QueryInterfaceTimeDimension(ABC):
+    """Time Dimension in a where clause with Jinja."""
+
+    def __init__(  # noqa
+        self,
+        time_dimension_name: str,
+        time_granularity_name: str,
+        entity_path: Sequence[str] = (),
```

Comment:
Are these meant to be accessible properties? It seems like we'd want to access them. Assuming that's the case, I recommend something more like the protocol definitions we use in dbt-semantic-interfaces, with property accessor methods defined on the protocol and everything else left up to the implementing classes.

Reply:
I don't think this still applies, given my recent refactors. Let me know  if that isn't the case

Reply:
Yeah I don't think this comment makes sense anymore.